beacon_node/beacon_chain/src/validator_monitor.rs

```diff
@@ -487,8 +500,66 @@ impl<T: EthSpec> ValidatorMonitor<T> {
                         u64_to_i64(validator.withdrawable_epoch),
                     );
                 }
+
+                // Add non finalized missed blocks for the monitored validators for the current slot of the current epoch
+                let current_slot = state.slot();
+                self.missed_blocks.contains(&(current_epoch, i as u64, current_slot))
+                    .then(|| {
+                        metrics::inc_counter_vec(
+                            &metrics::VALIDATOR_MONITOR_MISSED_NON_FINALIZED_BLOCKS_TOTAL,
+                            &[current_epoch.to_string().as_str(), current_slot.to_string().as_str(), id],
+                        );
+                    });
+            }
+        }
+    }
+
+    fn add_validators_missed_blocks(&mut self, current_epoch: Epoch, state: &BeaconState<T>) {
+        // Determine missed (non-finalized) blocks (can contain false positives)
+        let range_of_slot = (T::slots_per_epoch() as usize - MISSED_BLOCK_LAG_SLOTS) - 1;
+        for n in 0..range_of_slot {
+            let slot = state.slot() - Slot::new(n as u64);
+            let prev_slot = slot - 1;
+
+            // safeguard if we are on a new chain, the slot - 1 can be negative
+            if prev_slot < 0 {
+                break;
+            }
+
+            // condition for missed_block is defined such as block_root == block_root - n
+            // where the proposer who missed the block is the proposer of the block at block_root - n
```

Comment:
This code is out of sync with the code (which is correct). The proposer who missed is the one at `slot` where `block_root(slot) == block_root(slot - 1)`.